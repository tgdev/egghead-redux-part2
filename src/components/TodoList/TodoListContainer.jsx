import { connect } from 'react-redux';

import { toggleTodo } from 'actions/todo.actions';

import getVisibleTodos from 'utils/get-visible-todos';

import TodoList from 'components/TodoList/TodoList';

const mapStateToProps = state => {
  return {
    todos: getVisibleTodos(state.todos, state.visibilityFilter)
  };
};

const mapDispatchToProps = dispatch => {
  return {
    handleTodos: id => {
      dispatch(toggleTodo(id));
    }
  };
};

const TodoListContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(TodoList);

export default TodoListContainer;
