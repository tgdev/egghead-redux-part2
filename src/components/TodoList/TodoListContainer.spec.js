import React from 'react';
import { shallow } from 'enzyme';

import appStore from '../../store';
import TodoListContainer from './TodoListContainer';

describe('TodoListContainer component', () => {
  const storeContext =  { context: { store: appStore } };
  it('renders TodoListContainer correctly', () => {
    const wrapper = shallow(<TodoListContainer />, storeContext);
    expect(wrapper).toMatchSnapshot();
  });
});
