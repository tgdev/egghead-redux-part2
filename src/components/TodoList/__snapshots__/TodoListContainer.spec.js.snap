// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListContainer component renders TodoListContainer correctly 1`] = `
<TodoList
  handleTodos={[Function]}
  todos={Array []}
/>
`;
